lib/core/dns-resolver.js

```diff
@@ -266,6 +266,10 @@ class DNSResolver {
       ignoreNoResultErrors(this._resolve6(hostname, ttl))
     ])
 
+    if (entries[0].status === 'rejected' && entries[1].status === 'rejected') {
+      throw entries[0].reason
+    }
```

Comment:
Why only throw the first error? Wouldn't it be better to throw an AggregateError that contains both?

Reply:
doing that, using AggregateError and modifying tests